proposed/http-middleware/middleware-meta.md

```diff
@@ -171,6 +170,33 @@ that are based around StackPHP.
 The double pass approach is much newer but has already been widely adopted by
 early adopters of HTTP Messages.
 
+
+### 4.4 Chosen Approach
+
+The double-pass approach has some significant issues regarding implementation.
+
+The most severe is that passing an empty response has no guarantees that the
+response is in a usable state. This is further exacerbated by the fact that a
+middleware may modify the response before passing it for further dispatching.
+
+Further compounding the problem is that there is no way to ensure that the
+response body has not been written to, which can lead to incomplete output or
+error responses being sent with cache headers attached.
```

Comment:
Another issue, as detailed in a [blog post by Rob Allen](https://akrabat.com/filtering-the-psr-7-body-in-middleware/), is that it's difficult to get the response *body* into a clean state if it has been previously written to; rewinding and writing back to it can lead to issues if the new content is of shorter length than the previous. As such, developers currently must depend on a concrete `StreamInterface`, populate it, and pass it to generate a new response.